src/parse/parser_public.ts

```diff
@@ -280,15 +283,25 @@ export class Parser {
 
             let orgProductions = this._productions
             let clonedProductions = new HashTable<Rule>()
-            // clone the grammar productions to support grammar inheritance. requirements:
-            // 1. We want to avoid rebuilding the grammar every time so a cache for the productions is used.
-            // 2. We need to collect the production from multiple grammars in an inheritance scenario during constructor invocation
-            //    so the myGast variable is used.
-            // 3. If a Production has been overridden references to it in the GAST must also be updated.
-            forEach(orgProductions.keys(), key => {
-                let value = orgProductions.get(key)
-                clonedProductions.put(key, cloneProduction(value))
-            })
+            if (serializedGrammar === undefined || serializedGrammar === null) {
```

Comment:
maybe sufficient to use idiomatic JavaScript

> if (!obj) { ... }

I use more specific conditions (=== null) in hot spots where there is a performance consideration